crates/chain/src/local_chain.rs

```diff
@@ -545,31 +545,44 @@ impl std::error::Error for ApplyHeaderError {}
 
 /// Applies `update_tip` onto `original_tip`.
 ///
-/// On success, a tuple is returned `(changeset, can_replace)`. If `can_replace` is true, then the
-/// `update_tip` can replace the `original_tip`.
+/// On success, a tuple is returned ([`CheckPoint`], [`ChangeSet`]).
```

Comment:
```suggestion
/// On success, a tuple is returned ([`CheckPoint`], [`ChangeSet`]). A [`CannotConnectError`] occurs when the `original_tip` and `update_tip` chains are disjoint.
```

I'd move the rest of the explanation for `CannotConnectError` to that type's doc. 

But I don't want to bike-shed it too much so I'll also ACK this as is and you decide.